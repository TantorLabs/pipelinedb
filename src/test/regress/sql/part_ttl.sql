CREATE FOREIGN TABLE ttl_stream (x integer) SERVER pipelinedb;

CREATE VIEW ttl0 WITH (ttl='3 seconds', ttl_column='ts',
                       partition_by='ts', partition_duration='1 second')
	AS SELECT arrival_timestamp AS ts, x FROM ttl_stream;

SELECT c.relname, ttl, ttl_attno FROM pipelinedb.cont_query cq
JOIN pg_class c ON c.oid = cq.relid WHERE c.relname = 'ttl0'
ORDER BY c.relname;

INSERT INTO ttl_stream (x) VALUES (0);
INSERT INTO ttl_stream (x) VALUES (1);
INSERT INTO ttl_stream (x) VALUES (2);

SELECT x, "$pk" FROM ttl0_mrel ORDER BY ts;

SELECT pg_sleep(3);

SELECT 0 * pipelinedb.ttl_expire('ttl0');

SELECT x, "$pk" FROM ttl0_mrel ORDER BY ts;

INSERT INTO ttl_stream (x) VALUES (0);
INSERT INTO ttl_stream (x) VALUES (1);
INSERT INTO ttl_stream (x) VALUES (2);

SELECT x, "$pk" FROM ttl0_mrel ORDER BY ts;

SELECT pg_sleep(3);
SELECT 0 * pipelinedb.ttl_expire('ttl0');

SELECT x, "$pk" FROM ttl0_mrel ORDER BY ts;

DROP VIEW ttl0;

CREATE VIEW ttl2 WITH (partition_by='ts', partition_duration='1 second')
    AS SELECT arrival_timestamp AS ts, x FROM ttl_stream;

DROP VIEW ttl2;

DROP FOREIGN TABLE ttl_stream CASCADE;
