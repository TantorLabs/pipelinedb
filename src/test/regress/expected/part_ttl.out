CREATE FOREIGN TABLE ttl_stream (x integer) SERVER pipelinedb;
CREATE VIEW ttl0 WITH (ttl='3 seconds', ttl_column='ts',
                       partition_by='ts', partition_duration='1 second')
	AS SELECT arrival_timestamp AS ts, x FROM ttl_stream;
SELECT c.relname, ttl, ttl_attno FROM pipelinedb.cont_query cq
JOIN pg_class c ON c.oid = cq.relid WHERE c.relname = 'ttl0'
ORDER BY c.relname;
 relname | ttl | ttl_attno 
---------+-----+-----------
 ttl0    |   3 |         1
(1 row)

INSERT INTO ttl_stream (x) VALUES (0);
INSERT INTO ttl_stream (x) VALUES (1);
INSERT INTO ttl_stream (x) VALUES (2);
SELECT x, "$pk" FROM ttl0_mrel ORDER BY ts;
 x | $pk 
---+-----
 0 |   1
 1 |   2
 2 |   3
(3 rows)

SELECT pg_sleep(3);
 pg_sleep 
----------
 
(1 row)

SELECT 0 * pipelinedb.ttl_expire('ttl0');
 ?column? 
----------
        0
(1 row)

SELECT x, "$pk" FROM ttl0_mrel ORDER BY ts;
 x | $pk 
---+-----
 0 |   1
 1 |   2
 2 |   3
(3 rows)

INSERT INTO ttl_stream (x) VALUES (0);
INSERT INTO ttl_stream (x) VALUES (1);
INSERT INTO ttl_stream (x) VALUES (2);
SELECT x, "$pk" FROM ttl0_mrel ORDER BY ts;
 x | $pk 
---+-----
 0 |   1
 1 |   2
 2 |   3
 0 |   4
 1 |   5
 2 |   6
(6 rows)

SELECT pg_sleep(3);
 pg_sleep 
----------
 
(1 row)

SELECT 0 * pipelinedb.ttl_expire('ttl0');
 ?column? 
----------
        0
(1 row)

SELECT x, "$pk" FROM ttl0_mrel ORDER BY ts;
 x | $pk 
---+-----
 0 |   1
 1 |   2
 2 |   3
 0 |   4
 1 |   5
 2 |   6
(6 rows)

DROP VIEW ttl0;
CREATE VIEW ttl2 WITH (partition_by='ts', partition_duration='1 second')
    AS SELECT arrival_timestamp AS ts, x FROM ttl_stream;
DROP VIEW ttl2;
DROP FOREIGN TABLE ttl_stream CASCADE;
